crates/utils/re_uri/src/redap_uri.rs

```diff
@@ -401,65 +401,101 @@ mod tests {
 
     #[test]
     fn test_proxy_endpoint() {
-        let url = "rerun://localhost:51234/proxy";
+        let url = "rerun://localhost:9876/proxy";
         let address: Result<RedapUri, _> = url.parse();
 
         let expected = RedapUri::Proxy(ProxyUri {
             origin: Origin {
                 scheme: Scheme::Rerun,
                 host: url::Host::Domain("localhost".to_owned()),
-                port: 51234,
+                port: 9876,
             },
         });
 
         assert_eq!(address.unwrap(), expected);
 
-        let url = "rerun://localhost:51234/proxy/";
+        let url = "rerun://localhost:9876/proxy/";
         let address: Result<RedapUri, _> = url.parse();
 
         assert_eq!(address.unwrap(), expected);
     }
 
     #[test]
     fn test_catalog_default() {
-        let url = "rerun://localhost:51234";
+        let url = "rerun://localhost:9876";
         let address: Result<RedapUri, _> = url.parse();
 
         let expected = RedapUri::Catalog(CatalogUri {
             origin: Origin {
                 scheme: Scheme::Rerun,
                 host: url::Host::Domain("localhost".to_owned()),
-                port: 51234,
+                port: 9876,
             },
         });
 
         assert_eq!(address.unwrap(), expected);
 
-        let url = "rerun://localhost:51234/";
+        let url = "rerun://localhost:9876/";
         let address: Result<RedapUri, _> = url.parse();
 
         assert_eq!(address.unwrap(), expected);
     }
 
     #[test]
-    #[ignore]
-    // TODO(lucasmerlin): This should ideally work but doesn't right now because of a issue in the `url` crate:
-    // https://github.com/servo/rust-url/issues/957
-    // See also `replace_and_parse` in `origin.rs`
     fn test_default_port() {
         let url = "rerun://localhost";
 
         let expected = RedapUri::Catalog(CatalogUri {
             origin: Origin {
                 scheme: Scheme::Rerun,
                 host: url::Host::Domain("localhost".to_owned()),
-                port: 51234,
+                port: 9876,
             },
         });
 
         assert_eq!(url.parse::<RedapUri>().unwrap(), expected);
     }
 
+    #[test]
+    fn test_default_everything() {
+        let test_cases = [
+            (
+                "localhost",
+                RedapUri::Catalog(CatalogUri {
+                    origin: Origin {
+                        scheme: Scheme::RerunHttp,
+                        host: url::Host::Domain("localhost".to_owned()),
+                        port: 9876,
+                    },
+                }),
+            ),
```

Comment:
That's about connecting to a local server right? Is the server switching to 9876 as well?